core/src/main/java/com/google/instrumentation/trace/Samplers.java

```diff
@@ -98,6 +102,7 @@ public String toString() {
     }
   }
 
+  @Immutable
   private static final class ProbabilitySampler extends Sampler {
```

Comment:
It's up to you.  I just suggested AutoValue as a way to avoid boilerplate.  I didn't think of it as documenting the purpose of the class.